### ISSUE TITLE
[BUG] broken pipe using shell redirections

### ISSUE THREAD

**Issue by NyaaaWhatsUpDoc** [raphamorim/rio#702]
for example i get the following every time, without fail:

```
$ tr -cd 'a-z' < /dev/urandom | head -c 10
ymbrconkfrtr: write error: Broken pipe
```

**Comment by raphamorim** [comment]
Hey @NyaaaWhatsUpDoc thanks for the issue, I am unable to reproduce it.

<img width="912" alt="Screenshot 2024-10-09 at 19 47 28" src="https://github.com/user-attachments/assets/e36759f5-f5bd-40fb-96be-8e652f1ae236">

I get same result in alacritty and kitty

<img width="847" alt="Screenshot 2024-10-09 at 19 48 21" src="https://github.com/user-attachments/assets/93a0710f-7f92-4b94-b682-1ec66fd85823">

**Comment by NyaaaWhatsUpDoc** [comment]
oh right you're on macOS, in which case try setting `export LC_CTYPE=C` before running the above (that will be why tr: is throwing a byte sequence error)

**Comment by raphamorim** [comment]
It works for me <img width="653" alt="Screenshot 2024-10-09 at 20 05 43" src="https://github.com/user-attachments/assets/1abeaca3-3f2c-4458-93ca-efa03deb70ae">

**Comment by NyaaaWhatsUpDoc** [comment]
let me just enable debug logging and see if i can reveal anything

**Comment by raphamorim** [comment]
Hey @NyaaaWhatsUpDoc did you find anything?

i still can’t repro 🙏

**Comment by NyaaaWhatsUpDoc** [comment]
this seems to be fixed in v0.1.17 it seems :) version of rio on Void is currently v0.1.16 which is what i was seeing this with. i'm going to submit an update to void-packages to update theirs
